source/Metadata/Javadoc.cpp

```diff
@@ -118,38 +135,39 @@ postProcess()
 find_brief:
     while(it != blocks_.end())
     {
-        if(it->kind == doc::Kind::brief)
+        auto& block_ptr = *it;
+        if(block_ptr->kind == doc::Kind::brief)
         {
-            brief = static_cast<doc::Paragraph*>(&*it);
+            brief = static_cast<doc::Paragraph*>(block_ptr.get());
             break;
         }
-        else if(it->kind == doc::Kind::param)
+        else if(block_ptr->kind == doc::Kind::param)
         {
-            it = blocks_.move_to(it, params_);
+            it = move_to(params_, blocks_, it);
             continue;
         }
-        else if(it->kind == doc::Kind::tparam)
+        else if(block_ptr->kind == doc::Kind::tparam)
         {
-            it = blocks_.move_to(it, tparams_);
+            it = move_to(tparams_, blocks_, it);
             continue;
         }
         ++it;
     }
 done:
     if(brief != nullptr)
     {
-        brief_ = blocks_.extract_first_of<doc::Paragraph>(
-            [brief](doc::Block& block)
+        auto match = std::find_if(blocks_.begin(), blocks_.end(),
+            [brief](std::unique_ptr<doc::Block>& block)
             {
-                return brief == &block;
+                return brief == block.get();
             });
+        MRDOX_ASSERT(match != blocks_.end());
+        brief_.reset(static_cast<doc::Paragraph*>(
+            match->release()));
+        blocks_.erase(match);
     }
-    else
-    {
-        static std::shared_ptr<doc::Paragraph const> empty_para =
-            std::make_shared<doc::Paragraph>();
-        brief_ = empty_para;
-    }
+    // KRYSTIAN FIXME: should an empty paragraph be used
+    // as the brief when no written brief exists?
```

Comment:
My thinking is that the generator wants to know that the brief was left out, in order to have more information when rendering. But it is also a pain in the ass because the generator has to check "if( ! brief )` all the time, which is a hassle in the Handlebars template. We need more practice/experience writing generators to know the answer to this.